### ISSUE TITLE
Build errors on 10.10.5

### ISSUE THREAD

**Issue by erco77** [fltk/fltk#501]
Using current fltk 1.4.x (git commit: debbba19), used the following to build:
`./configure --enable-localjpeg --enable-localpng --enable-localzlib ; make clean; make -j 4`

Getting these fatal errors building on MacOS Yosemite (10.10.5):
```
Compiling Fl_cocoa.mm...
Fl_cocoa.mm:1391:8: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
      [Fl_X::i(w)->xid makeKeyWindow];
       ^~~~~~~~~~~~~~~
Fl_cocoa.mm:1565:28: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
    if ( !w->parent() && ![x->xid isMiniaturized]) {
                           ^~~~~~
Fl_cocoa.mm:2944:43: error: use of undeclared identifier 'NSOpenGLContextParameterSurfaceOpacity'
  [ctx getValues:&gl_opacity forParameter:NSOpenGLContextParameterSurfaceOpacity];
                                          ^
Fl_cocoa.mm:2947:45: error: use of undeclared identifier 'NSOpenGLContextParameterSurfaceOpacity'
    [ctx setValues:&gl_opacity forParameter:NSOpenGLContextParameterSurfaceOpacity];
                                            ^
Fl_cocoa.mm:3338:6: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
    [i->xid setMinSize:minSize];
     ^~~~~~
Fl_cocoa.mm:3339:6: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
    [i->xid setMaxSize:maxSize];
     ^~~~~~
Fl_cocoa.mm:3370:43: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
  if (!shown() && (!parent() || (top && ![top->xid isMiniaturized]))) {
                                          ^~~~~~~~
Fl_cocoa.mm:3375:12: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
      if ([i->xid isMiniaturized]) {
           ^~~~~~
Fl_cocoa.mm:3377:10: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
        [i->xid deminiaturize:nil];
         ^~~~~~
Fl_cocoa.mm:3380:10: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
        [i->xid makeKeyAndOrderFront:nil];
         ^~~~~~
Fl_cocoa.mm:4195:30: warning: receiver type 'fl_uintptr_t' (aka 'unsigned long') is not 'id' or interface pointer, consider casting it to 'id' [-Wreceiver-expr]
  FLView *myview = (FLView*)[Fl_X::i(win)->xid contentView];
                             ^~~~~~~~~~~~~~~~~
9 warnings and 2 errors generated.
make[1]: *** [Fl_cocoa.o] Error 1
make: *** [all] Error 1
```

**Comment by erco77** [comment]
Looks like the first of the warnings was introduced in c720aae515.
Then the fatal error was introduced in commit 13e05f4204c.

**Comment by ManoloFLTK** [comment]
@erco77 That should be fixed with 8bbedd6 . Please, close the issue if you confirm.

**Comment by erco77** [comment]
Confirmed -- builds to completion now -- thanks Manolo!
